Controller/Webhook/Index.php

```diff
@@ -11,352 +11,135 @@
-        $merchantAccount = $this->configHelper->getMerchantAccount()
-            ?? $this->configHelper->getMotoMerchantAccounts();
-
-        $authResult = $this->notificationReceiver->isAuthenticated(
-            $response,
-            $merchantAccount,
-            $this->configHelper->getNotificationsUsername(),
-            $this->configHelper->getNotificationsPassword()
-        );
-
-        // if the number of wrongful attempts is not less than 6, save it in cache
-        if($this->rateLimiterHelper->getNumberOfAttempts() >= $this->rateLimiterHelper::NUMBER_OF_ATTEMPTS) {
-            $this->rateLimiterHelper->saveSessionIdIpAddressToCache();
-            return false;
+        if (isset($payload['eventCode'])) {
+            return WebhookAcceptorInterface::TYPE_STANDARD;
         }
 
-        // if there is no auth result, save it in cache
-        if(!$authResult) {
-            $this->rateLimiterHelper->saveSessionIdIpAddressToCache();
-            return false;
+        if (isset($payload['type']) && str_contains($payload['type'], 'token')) {
+            return WebhookAcceptorInterface::TYPE_TOKEN;
         }
 
-        return $authResult;
+        throw new \UnexpectedValueException('Unable to determine webhook type from payload.');
     }
 
-    /**
-     * save notification into the database for cronjob to execute notification
-     *
-     * @param $response
-     * @param $notificationMode
-     * @return bool
-     * @throws AuthenticationException
-     * @throws MerchantAccountCodeException
-     * @throws HMACKeyValidationException
-     * @throws InvalidDataException
-     */
-    private function processNotification(array $response, $notificationMode)
+    private function isNotificationModeValid(array $payload): bool
     {
-        if (!$this->isAuthorised($response)) {
-            return false;
-        }
-
-        // Validate if Ip check is enabled and if the notification comes from a verified IP
-        if (!$this->isIpValid()) {
-            $this->adyenLogger->addAdyenNotification(sprintf(
-                    "Notification has been rejected because the IP address could not be verified",
-                ),
-                [
-                    'pspReference' => $response['pspReference'],
-                    'merchantReference' => $response['merchantReference']
-                ]
-            );
-            return false;
-        }
-
-        // Validate the Hmac calculation
-        $hasHmacCheck = $this->configHelper->getNotificationsHmacKey() &&
-            $this->hmacSignature->isHmacSupportedEventCode($response);
-        if ($hasHmacCheck && !$this->notificationReceiver->validateHmac(
-            $response,
-            $this->configHelper->getNotificationsHmacKey()
-        )) {
-            $this->adyenLogger->addAdyenNotification(
-                'HMAC key validation failed ' . json_encode($response)
-            );
+        if (!isset($payload['live'])) {
```

Comment:
This block will always return `false` for non-standard webhooks as they do not have `live` field. So, it ends up with returning `401` HTTP response.